sound/soc/sof/ipc4-topology.h

```diff
@@ -211,6 +215,20 @@ struct sof_ipc4_control_data {
 	};
 };
 
+static inline int sof_ipc4_ctl_is_initinstance(struct snd_sof_control *scontrol)
+{
+	struct sof_ipc4_control_data *control_data = scontrol->ipc_control_data;
+	int *magic_num;
+
+	magic_num = control_data->data->data;
+
+	if (magic_num[0] != IPC4_KCONTROL_MAGIC_NUM[0])
+		return 0;
+	if (magic_num[1] != IPC4_KCONTROL_MAGIC_NUM[1])
+		return 0;
```

Comment:
We can combine the two tests together.
Like
```
if (magic_num[0] == IPC4_KCONTROL_MAGIC_NUM[0] && magic_num[1] == IPC4_KCONTROL_MAGIC_NUM[1])
        return 1;

return 0;
```

Reply:
WIll update it in next version.